src/daemon.cpp

```diff
@@ -31,6 +31,9 @@ Daemon::Daemon(QObject* parent): QObject(parent)
     QDBusConnection dbus = QDBusConnection::sessionBus();
     dbus.registerObject("/KDENow", this);
     dbus.registerService("org.kde.kdenow");
+    Processor * processor = SingletonFactory::instanceFor< Processor >();
+    connect(this, &Daemon::fetchRecordsFromDatabase, processor, &Processor::loadPlugins);
```

Comment:
No need to connect it via signal and emit the signal in next line. 

Just do, processor->loadPlugins();